lib/atlas/runner/set_attributes_from_graph_values.rb

```diff
@@ -0,0 +1,50 @@
+module Atlas
+  class Runner
+    module SetAttributesFromGraphValues
+      def self.with_dataset(dataset)
+        lambda do |refinery|
+          apply_graph_methods!(refinery, dataset)
+
+          refinery
+        end
+      end
+
+      private
+
+      def self.apply_graph_methods!(refinery, dataset)
+        refinery.nodes.each do |node|
+          set_graph_methods(dataset, node)
+
+          node.out_edges.each do |edge|
+            set_graph_methods(dataset, edge)
+          end
+        end
+      end
+
+      # Private: set_graph_methods
+      #
+      # el = Atlas element either a Slot, Edge or Node
+      # dataset = an Atlas::Datset::Derived
+      def self.set_graph_methods(dataset, refinery_element, atlas_element = nil)
+        atlas_element = atlas_element || refinery_element.get(:model)
+
+        (dataset.graph_values.for(atlas_element) || {})
+          .each_pair do |method, val|
+            if %w(input output).include?(method)
```

Comment:
Small suggestion: This creates three new objects for every initializer method (two strings and an array). You can eliminate that by adding a `# frozen_string_literal: true` comment to the top of the file and using `case`:

```ruby
case method
when 'input', 'output'
  set_graph_methods_to_slot!(refinery_element, method, val)
else
  refinery_element.set(method.to_sym, val)
end
```